userManagement/views.py

```diff
@@ -30,3 +32,69 @@ def settings(request):
         "p_form": p_form,
     }
     return render(request, 'userManagement/settings.html', context)
+
+
+def userAccount(request):
+    if request.method == 'GET':
+        if not request.user.is_superuser:
+            return render(request, "security/403.html", status=403)
+
+    if request.method == 'POST':
+        c_form = userCreationForm(request.POST)
+        if c_form.is_valid():
+            c_form.save()
+            messages.success(request, f'Account was successfully created!')
+            return redirect('user-account')
+    else:
+        c_form = userCreationForm(request.POST)
+
+    context = {
+        'staff': User.objects.all(),
+        'c_form': c_form,
+    }
+
+    return render(request, 'userManagement/userAccount.html', context)
+
+#TODO The page is to create a user account when user clicks the link in the invitation email
+def userRegister(request):
+    if request.method == 'POST':
+        c_form = userCreationForm(request.POST)
+        if c_form.is_valid():
+            c_form.save()
+            messages.success(request, f'Account was successfully created!')
+            return redirect('user-account')
+    else:
+        c_form = userCreationForm(request.POST)
+
+    context = {
+        'staff': User.objects.all(),
+        'c_form': c_form,
+    }
+
+    return render(request, 'userManagement/userAccount.html', context)
+
+
+def userDelete(request, email):
+    u = User.objects.filter(email=email).first()
+    u.groups.all().delete()
+    u.deleteFlag = True
+    u.save()
+    messages.success(request, f'Account {email} was successfully deleted!')
+
+    return redirect('user-account')
+
+def userAdmin(request, email):
```

Comment:
require staff annotation.
Needs a better more descriptive name.
Why do we make two entirely different calls for setting a role to userAdmin and userLabAssitant instead of just passing in a parameter (role)

Reply:
You just added @login_not_required to the url responsible for making a user an admin. This means anyone without an account can make any user admin